students/robert/Session 4/mailroom.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python 
+
+def main_command():
+    while True:
+        user_choice = input("Type 1 for Send a Thank You; Type 2 for Create Report: Type 3 to quit: ->")
+        try:
+            user_choice = int(user_choice)
+            if user_choice not in [1, 2, 3]:
+              print("Invalid selection. Please try again.")
+            else:
+              break
+        except:
+            print("Invalid selection. Please try again.")
+        finally:
+            pass
+    return user_choice
+
+
+
+def donor_name():
+    for i in dln.keys():
+        print(i) 
+
+def Sending_a_Thank_You():
+    prompt = True
+    while prompt: 
+        full_name = input("Full Name: |'list'| ->")
+        if full_name == 'list':
+           donor_name()
+           prompt = True
+        elif full_name == 'main':
+           break
+        elif full_name in dln.keys():
+           prompt = False
+        else:        
+            dln[full_name] = []
+            prompt = False
+    promptN = True
+    while promptN:
+        donation = input("Donation Amount: ->")
+        try:
+            val = int(donation)
+        except ValueError:
```

Comment:
It would be very helpful for the end-user if you added a statement here letting them know they need to enter an int for donation amount, instead of just returning them to the prompt.